File: SFDataApplicationTestC++/shadow/system/data/objects/SFVertex2fData.cpp

#include "SFVertex2fData.h"


namespace sf{

SFVertex2fData::SFVertex2fData() :SFVectorData(2){
	this->vertex.set2f(0.0f,0.0f);
	vertex.setArray(getFloatValues());
}

SFVertex2f SFVertex2fData::getVertex2f(){
	return vertex;
}


SFVectorData* SFVertex2fData::clone() {
	return new SFVertex2fData();
}
}

File: sfoglintegration/SF20Data/src/shadow/system/data/objects/SFVertex2fData.java
package shadow.system.data.objects;

import shadow.math.SFVertex2f;

public class SFVertex2fData extends SFVectorData{
	
	private SFVertex2f vertex;
	
	public SFVertex2fData() {
		super(2);
		vertex=new SFVertex2f(getFloatValues());
	}

	public SFVertex2f getVertex2f(){
		return vertex;
	}
	
	@Override
	public SFVectorData copyDataObject() {
		return new SFVertex2fData();
	}
}
